pwndbg/dbg/lldb.py

```diff
@@ -221,6 +263,100 @@ def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
 
         return LLDBValue(value)
 
+    @override
+    def vmmap(self) -> pwndbg.dbg_mod.MemoryMap:
+        regions = self.process.GetMemoryRegions()
+
+        pages = []
+        for i in range(regions.GetSize()):
+            region = lldb.SBMemoryRegionInfo()
+            assert regions.GetMemoryRegionAtIndex(
+                i, region
+            ), "invalid region despite being in bounds"
+
+            perms = 0
+            if region.IsReadable():
+                perms |= os.R_OK
+            if region.IsWritable():
+                perms |= os.W_OK
+            if region.IsExecutable():
+                perms |= os.X_OK
+
+            # LLDB doesn't actually tell us the offset of the mapped file, just
+            # whether it is mapped or not.
+            offset = 0 if not region.IsMapped() else 1
+
+            objfile = region.GetName()
+            if objfile is None:
+                # LLDB will sometimes give us overlapping ranges with no name.
+                # For now, we ignore them, since GDB does not show them.
+                continue
+
+            pages.append(
+                pwndbg.lib.memory.Page(
+                    start=region.GetRegionBase(),
+                    size=region.GetRegionEnd() - region.GetRegionBase(),
+                    flags=perms,
+                    offset=offset,
+                    objfile=objfile,
+                )
+            )
+
+        return LLDBMemoryMap(pages)
+
+    @override
+    def symbol_name_at_address(self, address: int) -> str | None:
+        addr = lldb.SBAddress(address, self.target)
+        ctx = self.target.ResolveSymbolContextForAddress(addr, lldb.eSymbolContextSymbol)
+
+        if not ctx.IsValid():
+            return None
+
+        if not ctx.symbol.IsValid():
+            return None
+
+        return ctx.symbol.name
+
+    @override
+    def arch(self) -> pwndbg.dbg_mod.Arch:
+        endian0 = self.process.GetByteOrder()
+        endian1 = self.target.GetByteOrder()
+
+        if endian0 == lldb.eByteOrderInvalid:
```

Comment:
What? Why? pls document when this happens etc.

Reply:
It's a fallback for when the target has a sensible architecture, but the process might not. I've added a comment explaining it.